### PR TITLE
Fix batch handling of empty partition key and empty row key

### PR DESCRIPTION
**Pull request by joelverhagen**
Resolve https://github.com/Azure/Azurite/issues/1234.

### CODE REVIEW

**Comment by joelverhagen** [comment]
@edwin-huber, I took a stab at https://github.com/Azure/Azurite/issues/1234. I saw you were assigned so I hope you haven't started working on it. If so, feel free to repurpose/toss out this PR 😃.

File: src/table/handlers/TableHandler.ts [hunk 1]
```diff
@@ -359,16 +365,21 @@ export default class TableHandler extends BaseHandler implements ITableHandler {
 
   public async mergeEntity(
     _table: string,
-    partitionKey: string,
+    partitionKey: string | undefined,
```
Comment (edwin-huber) [inline src/table/handlers/TableHandler.ts:368]: @XiaoningLiu extending the parameter in the function signature with undefined is valid based on the "Duck Typing" in Typescript.
Are you happy with this?

**Comment by edwin-huber** [comment]
Thanks for the submission!
I like the changes; I have one small suggestion and would like @XiaoningLiu to weigh in on the Handler Method signature.

File: src/table/handlers/TableHandler.ts (hunk 1 above)
Comment (joelverhagen) [reply src/table/handlers/TableHandler.ts:368]: I think empty string is not a good indicator of missing value. An empty string is actually a valid partition key and `undefined` means the value should be picked from the context if possible.

**Review by XiaoningLiu — APPROVED** [review]

**Comment by XiaoningLiu** [comment]
@edwin-huber Do you have other comments before merging this PR?

**Review by edwin-huber — CHANGES_REQUESTED** [review]
Please collect all checks for values into a single function.
This will help us when we update validation logic in the future.

```
partitionKey = partitionKey !== undefined ? partitionKey : tableContext.partitionKey!;
```

File: src/table/handlers/TableHandler.ts [hunk 2]
```diff
@@ -275,16 +275,22 @@ export default class TableHandler extends BaseHandler implements ITableHandler {
...
     const table = this.getAndCheckTableName(tableContext);
-    partitionKey = partitionKey || this.getAndCheckPartitionKey(tableContext);
-    rowKey = rowKey || this.getAndCheckRowKey(tableContext);
+
+    partitionKey = partitionKey !== undefined ? partitionKey : tableContext.partitionKey!; // Get partitionKey from context
+    rowKey = rowKey !== undefined ? rowKey : tableContext.rowKey!; // Get rowKey from context
+
+    if (partitionKey === undefined || rowKey === undefined) {
+      throw StorageErrorFactory.getPropertiesNeedValue(context);
+    }
```
[the remark this answers is not in the document]
Comment (joelverhagen) [reply src/table/handlers/TableHandler.ts]: Done.

**Review by edwin-huber — APPROVED** [review]

**Comment by joelverhagen** [comment]
Are there any other things I should address before the merge? Happy to fix whatever 😃

**Pull request merged**